### PR TITLE
v1.9.1 geoips - VERSION RELEASE main - doc outline, bug fixes

### PR DESCRIPTION
**Pull request by mindyls**
# Related Issues
required to close NRLMMD-GEOIPS/geoips#171

# Reviewer Instructions
* Please confirm merge is into 'main' branch from 'v1.9.1-release'
* Review "Files changed" tab
* Confirm appropriate testing was completed for these updates.

# Summary
Changes for version 1.9.1 update on repo geoips.
* Remove GEOIPS_BASEDIR references

  * Add GEOIPS_TESTDATA_DIR and GEOIPS_DEPEDNENCIES_DIR to filenames/base_paths.py
* Sphinx Documentation Updates

  * Finalize RST documentation outline to match sharepoint doc
  * Update some documentation links
  * Implement sphinx-apidoc for documentation build
* Add missing fields in Rain colormap mpl_color_info
* Allow processing if no TCs in database
* Install geoips[doc,test,lint] in setup.sh
* Only add product to database if returned product is not None
* Replace <output> in single_source procflow in addition to <product>

  * This matches functionality in config_based.py
* Ignore flake8 W503 (conflicts with black)

See NRLMMD-GEOIPS/geoips#171 for additional repositories included in this update.

# Testing Instructions

Install GEOIPS and clone v1.9.1-release branch for:
* geoips
* recenter_tc
* .github
* geoips_clavrx

Tests should return 0:
$GEOIPS_PACKAGES_DIR/geoips/tests/test_full_install.sh

# Output

# Post Merge Steps
Once this PR has been merged, v1.9.1 will be tagged/released on the geoips main branch.

### CODE REVIEW

**Review by cpcamacho — APPROVED** [review]

File: docs/source/devguide/contributors.rst [hunk 1]
```diff
@@ -25,20 +25,16 @@ else may have time to finalize for approval even if you are unable.
 
 Contributions generally follow the process of:
 
-1. `Submit a GitHub Issue`_ for a bug fix or feature request.
-2. `Branch off a given Issue`_ in order to make the required changes
-3. `Create a github Pull Request`_ in order for your changes to be reviewed
-   prior to being merged to the integration branch
+1. Submit a GitHub Issue for a bug fix or feature request. :ref:`create_issue`
```
Comment (lance-wilson) [inline docs/source/devguide/contributors.rst:28]: Are these ":ref:" constructs supposed to link somewhere?  It doesn't appear that they do.

Comment (mindyls) [reply docs/source/devguide/contributors.rst:28]: Only when you actually build the documentation - they are sphinx directives.  We went back and forth on github native RST rendering (which do not link in HTML documentation), and sphinx directives (which are not clickable from within github).  Thoughts?

Comment (lance-wilson) [reply docs/source/devguide/contributors.rst:28]: I suppose it's probably better to have it in the built documentation if you only have one.  Is there anyway to have both links without it looking weird?

Comment (mindyls) [reply docs/source/devguide/contributors.rst:28]: I think we will have to address this through a separate issue - I am going to resolve this for now.

**Review by jsolbrig — APPROVED** [review]
Looks good to me!

**Pull request merged**